Fix build error "conflicting types for 'UTP_Write'"

Fetch from: http://www.adminsehow.com/2012/10/how-to-install-transmission-2-71-on-debian-6-0-squeeze/

The corresponding upstream bug report (unfixed) can be found here:
https://trac.transmissionbt.com/ticket/5232

Signed-off-by: Bernd Kuhls <berndkuhls@hotmail.com>

--- transmission-2.82.org/third-party/libutp/utypes.h	2013-08-09 04:47:43.000000000 +0200
+++ transmission-2.82/third-party/libutp/utypes.h	2014-01-02 20:17:18.000000000 +0100
@@ -35,8 +35,15 @@
 typedef const char * cstr;
 typedef char * str;
 
-#ifndef __cplusplus
-typedef uint8 bool;
-#endif
+//#ifndef __cplusplus
+//typedef uint8 bool;
+//#endif
+#ifndef __cplusplus
+#ifdef HAVE_STDBOOL_H
+#include <stdbool.h>
+#else
+typedef uint8 bool;
+#endif
+#endif
 
 #endif //__UTYPES_H__
